### PR TITLE
Refine Python DirectRunner watermark advancement behavior

### PR DESCRIPTION
**Pull request by charlesccychen**
This change helps prepare for streaming pipeline execution.

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @aaltay

File: sdks/python/apache_beam/runners/direct/watermark_manager.py [hunk 1]
```diff
@@ -193,9 +194,14 @@ def remove_pending(self, completed):
...
     with self._lock:
-      pending_holder = (WatermarkManager.WATERMARK_NEG_INF
-                        if self._pending else
-                        WatermarkManager.WATERMARK_POS_INF)
+      pending_holder = WatermarkManager.WATERMARK_POS_INF
+      for input_bundle in self._pending:
+        # TODO(ccy): we can have the Bundle class keep track of the minimum
+        # timestamp so we don't have to do an iteration here.
+        for wv in input_bundle.get_elements_iterable():
+          advanceable_timestamp = wv.timestamp - TIME_GRANULARITY
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/direct/watermark_manager.py]: What is the reason for `- TIME_GRANULARITY` ?

File: sdks/python/apache_beam/utils/timestamp.py [hunk 2]
```diff
@@ -208,3 +208,6 @@ def __rmul__(self, other):
   def __mod__(self, other):
     other = Duration.of(other)
     return Duration(micros=self.micros % other.micros)
+
+
+TIME_GRANULARITY = Duration(micros=1)
```
Comment (aaltay) [inline sdks/python/apache_beam/utils/timestamp.py]: Would this be relevant outside of DirectRunner?

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**